docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+
+The following steps explains the sequence of events: 
+
+1. The user enters `get patient S7777777Z` .
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`. 
+
+3. `PatientParser#parse()` will create an object `GetPatientCommand`.
+
+5. `HappyPills` then calls `GetPatientCommand.execute()` to execute the method. 
+
+6. `GetPatientCommand.execute()` calls `TextUi.getPatientSuccessMessage()` if a patient of NRIC `S7777777Z` can be found 
+or `TextUi.patientNotExists()` if no patient with the given NRIC exists in the program. 
+
+7. The respective `TextUi` methods will then display the message to the user accordingly.
+
+### 4.2 Patient Medical Records Features 
+
+
+### 4.3. Appointment Scheduling Feature 
+
+#### 4.3.1. Add Appointment
+
+The user is able to add appointments into the program to manage the appointment schedule. 
+The command: 
+
+    add appt /ic S7777777Z /d 01/04/2020 /t 12:00:00 /r Checkup 
+    
+will add an appointment with `NRIC` as S7777777Z with the following attributes: 
+* appointment date: `01/04/2020`
+* appointment time: `12:00:00`
+* reason for appointment: `Checkup`
+
+An `appointmentId` will also be given when an appointment is successfully added. 
+
+**Implementation** 
+
+![Add Appointment Sequence Diagram](images/AddAppointmentSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `add appt /ic S7777777Z /d 01/04/2020 /t 12:00:00 /r Checkup` .
+
+2. `HappyPills` calls `Parser#parse()` which then calls `AppointmentParser#parse()`. 
+
+3. `AppointmentParser#parse()` will create an object `AddAppointmentCommand`.
+
+5. `HappyPills` then calls `AddAppointmentCommand.execute()` to execute the method. 
+
+6. In `AddAppointmentCommand.execute()`, an appointment is added and the display message is returned.
+
+#### 4.3.2. Edit Appointment 
+
+#### 4.3.3. Delete Appointment 
+
+The user can delete an appointment from the list of appointments currently in the program. The command: 
+
+    delete appt 1 
+    
+will delete the appointment with appointment ID `1`, if found. 
+
+**Implementation** 
+
+The following steps explains the sequence of events: 
+
+1. The user enters `delete appt 1`. 
+
+2. `HappyPills` calls `Parser#parse()` which then calls `AppointmentParser#parse()`.
+
+3. `AppointmentParser#parse()` will then create an object `DeleteAppointmentCommand`. 
+
+4. `HappyPills` then calls `DeleteAppointmentCommand.execute()` to execute the command. 
+
+5. In `DeleteAppointmentCommand.execute()`, the respective patient is deleted and the display message is returned. 
+
+#### 4.3.4. Mark Appointment as Done 
+
+#### 4.3.5. List Appointments 
+
+The user is able to get a list of all the appointments currently in the program.
+The command: 
+    
+    list appt
+    
+will list all the appointments in the `AppointmentMap`. 
+
+**Implementation** 
+
+![List Appointment Sequence Diagram](images/ListAppointmentSequenceDiagram.jpg)
```

Comment:
Like in `images/AddPatientSequenceDiagram.jpg`, I suggest shortening the activation bars of the methods, as a method cannot remain active after it has returned.

From the CS2113/T website:
![bug1](https://user-images.githubusercontent.com/35903974/77999573-77fa3980-7365-11ea-8274-f6f23cffa7b2.png)